src/RageSoundMixBuffer.cpp

```diff
@@ -14,14 +14,27 @@ static bool g_bVector = Vector::CheckForVector();
 
 RageSoundMixBuffer::RageSoundMixBuffer()
 {
-	m_iBufSize = m_iBufUsed = 0;
-	m_pMixbuf = nullptr;
+	// Set m_iBufSize to 2MB.
+	m_iBufSize = 2 * 1024 * 1024 / sizeof(float);
+
+	// Allocate memory
+	m_pMixbuf = static_cast<float*>(std::malloc(m_iBufSize * sizeof(float)));
+
+	// Check if memory allocation was successful
+	if (m_pMixbuf == nullptr)
+	{
+		std::exit(EXIT_FAILURE);
```

Comment:
Haven't decided what the best course of action here is, not convinced `std::exit(EXIT_FAILURE)` is ideal

Reply:
Maybe ASSERT_M just indicating the you weren't able to allocate enough memory. Similar behavior but at least we'd get a log? I think it seems like a rare issue to run into considering we only need 2MB.